slice/internal/controller/workload_controller.go

```diff
@@ -457,38 +482,43 @@ func (r *WorkloadReconciler) syncSlices(ctx context.Context, wl *kueue.Workload,
 	return nil
 }
 
-func shouldCreateSliceForPodSetAssignment(wl *kueue.Workload, psa kueue.PodSetAssignment) bool {
+func shouldCreateSliceForPodSetAssignment(wl *kueue.Workload, psa kueue.PodSetAssignment, nodes map[string]corev1.Node) bool {
 	if podSet := podset.FindPodSetByName(wl.Spec.PodSets, psa.Name); podSet != nil {
-		return core.IsRelevantPodTemplateSpec(podSet.Template) && topology.IsAssignmentValid(psa)
+		return core.IsRelevantPodTemplateSpec(podSet.Template) && topology.IsAssignmentValid(psa, nodes)
 	}
 	return false
 }
 
-func parseTopologyAssignmentIntoNodeSelector(slice *v1alpha1.Slice, topologyAssignment *kueue.TopologyAssignment) {
+func parseTopologyAssignmentIntoNodeSelector(slice *v1alpha1.Slice, topologyAssignment *kueue.TopologyAssignment, nodes map[string]corev1.Node) {
 	nodeSelectors := sets.New[string]()
 	// we already validated that all assignments have a valid level,
 	// in validateRelevantWorkload.
-	subblockLevelIndex := topology.SubblockLevelIndex(topologyAssignment)
-	for _, domain := range topologyAssignment.Domains {
-		nodeSelectors.Insert(domain.Values[subblockLevelIndex])
+	if subblockLevelIndex := topology.LevelIndex(topologyAssignment, core.TPUSubBlockLabel); subblockLevelIndex != -1 {
+		for _, domain := range topologyAssignment.Domains {
+			nodeSelectors.Insert(domain.Values[subblockLevelIndex])
+		}
```

Comment:
Let's delete this branch, as I don't think we need it any more (and if I'm mistaken, we can restore it). I prefer to have just one well-lit path (then we won't have to update tests to cover both)

Reply:
However, deleting this branch would break the two-level topology. Do you think that's acceptable?

Reply:
Yes, because the 2-level topology doesn't work with the health labels properly (which is an important feature)

Reply:
Make sense.